frontend_lib/src/component/Tags/TagList.jsx

```diff
@@ -0,0 +1,180 @@
+      if (checkedToggled) {
+        putContentTag(props.apiUrl, props.workspaceId, props.contentId, tagId)
+      } else {
+        deleteContentTag(props.apiUrl, props.workspaceId, props.contentId, tagId)
+      }
+
+      return {
+        tagIsChecked: {
+          ...previousState.tagIsChecked,
+          [tagId]: checkedToggled
+        }
+      }
+    })
+  }
+
+  componentDidMount () {
+    this.updateTagList()
+  }
+
+  componentDidUpdate (prevProps) {
+    const { props } = this
+
+    if (prevProps.contentId !== props.contentId) {
+      this.updateTagList()
+    }
+  }
+
+  async updateTagList () {
+    const { props } = this
+    const fetchGetWsTagList = await handleFetchResult(
+      await getWorkspaceTagList(props.apiUrl, props.workspaceId)
+    )
+
+    if (!fetchGetWsTagList.apiResponse.ok) {
+      sendGlobalFlashMessage(props.t('Error while fetching a list of tags'))
+      return
+    }
+
+    const fetchGetContentTagList = await handleFetchResult(
+      await getContentTagList(props.apiUrl, props.workspaceId, props.contentId)
+    )
+
+    if (!fetchGetContentTagList.apiResponse.ok) {
+      sendGlobalFlashMessage(props.t('Error while fetching a list of tags'))
+    }
+
+    const isContentTag = (tag) => {
+      return fetchGetContentTagList.body.some(t => t.tag_id === tag.tag_id)
+    }
+
+    const sortTagList = (tagA, tagB) => {
+      const isTagAContent = isContentTag(tagA)
+      const isTagBContent = isContentTag(tagB)
+
+      if (!isTagAContent && isTagBContent) {
+        return 1
+      }
+
+      if (isTagAContent && !isTagBContent) {
+        return -1
+      }
+
+      return naturalCompare(tagA, tagB, props.i18n.language, 'tag_name')
+    }
+
+    const sortedTagList = fetchGetWsTagList.body.sort(sortTagList)
+
+    const tagIsChecked = {}
+
+    for (const tag of fetchGetContentTagList.body) {
+      tagIsChecked[tag.tag_id] = true
+    }
+
+    this.setState({ tagList: sortedTagList, tagIsChecked: tagIsChecked })
+  }
+
+  render () {
+    const { props, state } = this
+
+    return (
+      <div className='taglist' data-cy='taglist'>
+
+        <div className='taglist__header'>
+          {props.t('Tag List')}
+        </div>
+
+        <div className='taglist__wrapper'>
+          {props.displayNewTagForm
+            ? (
+              <NewTagForm
+                apiUrl={props.apiUrl}
+                workspaceId={props.workspaceId}
+                contentId={props.contentId}
+                onClickCloseAddTagBtn={props.onClickCloseAddTagBtn}
+              />
+            )
+            : (
+              <div className='taglist__btnadd' data-cy='taglist__btnadd' onClick={props.onClickAddTagBtn}>
+                <div className='taglist__btnadd__button primaryColorFontHover primaryColorBorderHover'>
+                  <div className='taglist__btnadd__button__plus'>
+                    <div className='taglist__btnadd__button__plus__icon'>
+                      <i className='fas fa-plus' />
```

Comment:
Can you use the `Icon` component? like that we guarantee that this icon has a tooltip (I suggest 'Add a tag')

Reply:
@gbonaspetti  I pushed modification, can you tell me if it's good for you ?

Reply:
Yes, just missing the title. See https://github.com/tracim/tracim/pull/4688#discussion_r649036173

Reply:
Fixed